lib/runner/extensions/event.command.js

```diff
@@ -453,6 +466,12 @@ module.exports = {
                         }
                     });
 
+                    // Handle errors from async script. If there was an error running the script itself,
+                    // that takes precedence
+                    if (!err) {
```

Comment:
Move this just after removing the event listeners.

The error needs to be serialized and all that is handled above.